### PR TITLE
#283 Allow multiple documents in a single post

### PR DESCRIPTION
**Pull request by stephanevieira75**
# Preview
Twake - Awesome chan

### CODE REVIEW

File: twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx [hunk 1]
```diff
@@ -40,7 +42,7 @@ export default (props: Props) => {
       props.onSend(content);
       return;
     }
-    if (content.trim()) {
+    if (content.trim() || messageEditorService.filesAttachements[props.threadId || 'main']) {
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx:45]: You must check the object is not empty (length) and the just if the object exists ( ([1, 2] == true) returns false)
Be careful and use a "?.length" (question mark in case it is not defined)

File: twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx [hunk 2]
```diff
@@ -67,18 +69,22 @@ export default (props: Props) => {
...
+          messageEditorService.getEditorId(props.threadId, props.messageId || '', props.context)
+        ) {
+          autocomplete.focus();
+        }
+        if (!message.parent_message_id) {
+          messageEditorService.openEditor(message.id, props.messageId || '');
+        }
+      })
+      .finally(() => {
+        messageEditorService.filesAttachements = {};
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx:86]: Here you removed all files from all editors (responses, main input etc) I suppose you want to empty only the files attached to the current editor. "messageEditorService.filesAttachements[threadId || "main"] = []"

File: twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx [hunk 3]
```diff
@@ -139,6 +145,7 @@ export default (props: Props) => {
       {!hasEphemeralMessage && !props.messageId && (
         <InputOptions
           inputValue={content}
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx:147]: It seems like you did not finish this one, create a "inputEmpty={!!content || hasFilesAttached}" props (hasFilesAttached is to define of course)

File: twake/frontend/src/app/scenes/Apps/Messages/Input/Parts/AttachedFiles.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        {editorManager.filesAttachements[props.threadId || 'main'].map((id: string) => {
+          return (
+            <File
+              mini
+              key={id}
+              data={{ id: id }}
+              notInDrive={true}
+              removeIcon
+              removeOnClick={(e: any) => {
+                e.stopPropagation();
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/scenes/Apps/Messages/Input/Parts/AttachedFiles.tsx:22]: Did you check that the preview was disabled when clicking outside the X button in this specific case ?
When the props "removeIcon" is true, I want the preview to be disabled.

File: twake/frontend/src/app/services/Apps/Messages/Messages.js [hunk 5]
```diff
@@ -142,6 +142,59 @@ class Messages extends Observable {
...
+
+      const preparedFiles = filesAttachementsToTwacode.length
+        ? [
+            filesAttachements.length >= 2
+              ? [...multipleFileSystemMessage]
+              : [...singleFileSystemMessage],
+            ...filesAttachementsToTwacode,
+          ]
+        : [];
+
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/services/Apps/Messages/Messages.js:196]: This is a lot of code, can we move it to a function ?

File: twake/frontend/src/app/components/Twacode/PseudoMarkdownDictionary.js [hunk 6]
```diff
@@ -113,13 +113,14 @@ class PseudoMarkdownDictionary {
...
-              style={{ marginBottom: 0 }}
-            />
-          </div>
+          <File
+            className="drive_view grid"
+            key={this.counter++}
+            data={{ id: object.content || '' }}
+            notInDrive={true}
+            mini={object.mode === 'mini' ? true : false}
+            style={{ marginBottom: 0, marginTop: 18 }}
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/components/Twacode/PseudoMarkdownDictionary.js:122]: It doesn't seems right the marginBottom: 18, ping me we have to look closer on this one together

File: twake/frontend/src/app/components/Drive/Drive.scss [hunk 7]
```diff
@@ -105,7 +105,7 @@
   flex-direction: column;
   line-height: 30px;
   margin-right: 16px;
-  margin-bottom: 16px;
+  margin-bottom: 4px !important;
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/components/Drive/Drive.scss:108]: You shouln't need to change the global Drive.scss UI (it can change the UI in the Drive app)
Please cancel changes to this file.

**Review by RomaricMourgues — CHANGES_REQUESTED** [review]

File: twake/frontend/src/app/services/languages/locale/en.js [hunk 8]
```diff
@@ -627,7 +626,8 @@ export default {
   'scenes.apps.account.languages.title': 'Language',
   'scenes.apps.account.languages.text': 'Change Twake language.',
   'scenes.apps.account.notifications.title': 'Notifications',
-  'scenes.apps.account.notifications.text': 'You can choose which notifications you want to receive here.',
+  'scenes.apps.account.notifications.text':
```
Comment (chamerling) [inline twake/frontend/src/app/services/languages/locale/en.js:629]: You should not change things which are not related to the current work. This is the same for some changes below

File: twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx (hunk 1 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx]: Done :+1:

File: twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx (hunk 2 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx]: Done :+1:

File: twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx (hunk 3 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx:153]: Done :+1:

File: twake/frontend/src/app/scenes/Apps/Messages/Input/Parts/AttachedFiles.tsx (hunk 4 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/scenes/Apps/Messages/Input/Parts/AttachedFiles.tsx]: Done :+1:

File: twake/frontend/src/app/services/Apps/Messages/Messages.js (hunk 5 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/services/Apps/Messages/Messages.js:186]: Done :+1:

File: twake/frontend/src/app/components/Drive/Drive.scss (hunk 7 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/components/Drive/Drive.scss]: Done :+1:

File: twake/frontend/src/app/services/Apps/Messages/Messages.js [hunk 9]
```diff
@@ -348,7 +390,12 @@ class Messages extends Observable {
 
   editMessage(messageId, value, messagesCollectionKey) {
     this.editedMessage = Collections.get('messages').find(messageId);
-    this.editedMessage.content = PseudoMarkdownCompiler.compileToJSON(value);
+    let content = PseudoMarkdownCompiler.compileToJSON(value);
+
+    content.prepared = [...this.editedMessage.content.prepared];
+    content.prepared[0] = value;
+
+    this.editedMessage.content = content;
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/services/Apps/Messages/Messages.js:398]: That seems strange to me, I don't understand how it can work :/

File: twake/frontend/src/app/scenes/Apps/Messages/Input/Parts/InputOptions.tsx [hunk 10]
```diff
@@ -168,8 +169,8 @@ export default (props: Props) => {
       )}
       <div style={{ flex: 1 }} />
       <div
-        className={'option ' + (props.inputValue ? '' : 'disabled ')}
-        onClick={() => props.onSend && props.onSend()}
+        className={'option ' + (props.inputValue || !props.isEmpty ? '' : 'disabled ')}
+        onClick={() => !props.isEmpty && props.onSend && props.onSend()}
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/scenes/Apps/Messages/Input/Parts/InputOptions.tsx:173]: Remove the props.inputValue because it is used nowhere else.
Only use the props.isEmpty

File: twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx [hunk 11]
```diff
@@ -139,6 +151,7 @@ export default (props: Props) => {
...
+          isEmpty={!!content || !hasFilesAttached}
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx:154]: This is wrong to me, the correct line should be
-> isEmpty={!content && !hasFilesAttached}
or
-> isEmpty={!(content || hasFilesAttached)}

**Review by RomaricMourgues — CHANGES_REQUESTED** [review]
Last fixes and it's done !

File: twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx (hunk 11 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/scenes/Apps/Messages/Input/Input.tsx:154]: Fixed

File: twake/frontend/src/app/scenes/Apps/Messages/Input/Parts/InputOptions.tsx (hunk 10 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/scenes/Apps/Messages/Input/Parts/InputOptions.tsx:173]: Removed :+1:

**Review by RomaricMourgues — APPROVED** [review]
